tests/gemm/test_bmm_bf16.py

```
@@ -65,6 +70,106 @@ def test_bmm_bf16(b, m, n, k, res_dtype, backend):
+        ((2, 8, 1024, 1024), torch.float32, 2),
+    ],
+)
+def test_bmm_bf16_weave_routes_and_output_identity(shape, res_dtype, expected_route):
+    if not bmm_bf16.is_backend_supported("weave", 103):
+        pytest.skip("Weave BF16 BMM requires exact SM103.")
+    if get_compute_capability(torch.device("cuda")) != (10, 3):
+        pytest.skip("Weave BF16 BMM requires exact SM103.")
+
+    b, m, n, k = shape
+    torch.manual_seed(7)
+    input = torch.randn((b, m, k), device="cuda", dtype=torch.bfloat16)
```

Ruff (0.16.1)

[error] Variable `input` is shadowing a Python builtin

(A001)